src/cartridges/int_adyen_SFRA/cartridge/controllers/middlewares/checkout/begin.js

```diff
@@ -8,6 +13,34 @@ function begin(req, res, next) {
     });
   }
 
+  // restore cart if order number was cached
+  try {
+    const cachedOrderNumber = req.session.privacyCache.get(
+      'currentOrderNumber',
+    );
+    if (cachedOrderNumber !== undefined) {
+      const currentBasket = BasketMgr.getCurrentBasket();
+      const currentOrder = OrderMgr.getOrder(cachedOrderNumber);
+      // if current basket is null or empty
+      if (
+        !currentBasket ||
+        currentBasket.getAllProductLineItems().length === 0
+      ) {
+        // if order status is CREATED we can fail it and restore basket
+        if (currentOrder.status.value === Order.ORDER_STATUS_CREATED) {
+          Transaction.wrap(() => {
+            currentOrder.trackOrderChange(
+              'Failing order so cart can be restored; Shopper navigated back to checkout during payment redirection',
+            );
+            OrderMgr.failOrder(currentOrder, true);
```

Comment:
Shall we add a logger here to explain what happened?

Reply:
the trackOrderChange adds a line in the order history. Do we need to put a Logger in for this as well?